### PR TITLE
Fro 46 channels

### PR DESCRIPTION
**Pull request by sir-ojay**
# Fixes Issue/Linear Ticket

( This could be an existing issue or a linear ticket )

**My PR Closes #issue_number_here** or
**My PR Fixes ID-46**

# Changes proposed

## What were you told to do?

1. Browse channels 

- See all channels with their descriptions
- Join any public channel
- Leave any Channel
- Search Bar for any Channel

2. Channel settings

- Mark all channels as read
- Show/ Hide all channel
- Sort Channels (By Recent, Priority, and Alphabetically) 
- Create a sidebar section Name of section
- Priority to be attached to the section
- Teams & Direct messages
- Create and Cancel Button
- Edit   

3. Channel groups

4. Add people to a channel group
5. Leave Group
6. Who can send messages to the group
7. Remove people from a group
8. Send Attachments

--

</body>
</html> -->

## What did you do?

# Check List (Check all the applicable boxes)

🚨Please review the [style guide for contributing](https://github.com/zurichat/zc_main/blob/dev/docs/STYLING.md) and [guidelines for contributing](https://github.com/zurichat/zc_main/blob/dev/docs/CONTRIBUTING.md) to this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [**not**] My code additions will fail neither code linting checks nor unit test.
- [x]
Screenshot (33)
] I 
Screenshot (34)
Screenshot (35)
am only making changes to files I was requested to.

# Screenshots/ Videos

### CODE REVIEW

**Comment by billmal071** [comment]
You have conflicts.

**Comment by JulianaSau** [comment]
All these things you created here already exist. All you had to do was edit them. Plus the UI design needs work. If you were using Slack as a reference you completely missed the mark😫

**Comment by JulianaSau** [comment]
They're located here https://github.com/zurichat/zc_main/blob/dev/packages/ui/src

File: package.json [hunk 1]
```diff
@@ -50,5 +50,8 @@
     "rimraf": "^3.0.2",
     "serve": "^12.0.0"
   },
-  "dependencies": {}
+  "dependencies": {
+    "@mui/material": "^5.10.16",
+    "react-hook-form": "^7.39.7"
```
Comment (ickynavigator) [inline package.json:55]: remove these from the root package

File: packages/main/src/components/channel/addpeople/AddPeople.jsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
+import styles from "./AddPeople.module.css";
+import cancel from "../../../assets/images/close.png";
+import * as React from "react";
+import { alpha, styled } from "@mui/material/styles";
+import { green } from "@mui/material/colors";
+import Switch from "@mui/material/Switch";
```
Comment (ickynavigator) [inline packages/main/src/components/channel/addpeople/AddPeople.jsx:6]: there's already a style library don't install a whole new one

File: packages/main/src/components/channel/channeldetails/ChannelDetails.jsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import styles from "./channeldetail.module.css";
+import cancel from "../../../assets/images/close.png";
+import star from "../../../assets/images/star.png";
+import calling from "../../../assets/images/call-calling.png";
+import notify from "../../../assets/images/notification.png";
+import copy from "../../../assets/images/copy.png";
```
Comment (ickynavigator) [inline packages/main/src/components/channel/channeldetails/ChannelDetails.jsx:8]: These are things you can get from react icons and style
you are bloating the app

File: packages/main/src/components/channel/channeldetails/ChannelDetails.jsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+            <p>About</p>
+          </Link>
+          <Link to="/" style={{ textDecoration: "none" }}>
+            <p>Members (14)</p>
+          </Link>
+          <Link to="/" style={{ textDecoration: "none" }}>
+            <p>Integrations</p>
+          </Link>
+          <Link to="/" style={{ textDecoration: "none" }}>
+            <p>Settings</p>
```
Comment (ickynavigator) [inline packages/main/src/components/channel/channeldetails/ChannelDetails.jsx:38]: why are all of these routes linking back to "/"?

File: packages/main/src/components/channel/createchannel/CreateChannel.jsx [hunk 5]
```diff
@@ -0,0 +1,179 @@
+import React, { useState, useEffect } from "react";
+import cancel from "../../../assets/images/close.png";
+import question from "../../../assets/images/question.png";
+import { alpha, styled } from "@mui/material/styles";
+import { green } from "@mui/material/colors";
+import Switch from "@mui/material/Switch";
+import { useForm } from "react-hook-form";
```
Comment (ickynavigator) [inline packages/main/src/components/channel/createchannel/CreateChannel.jsx:7]: formik is already installed
this package is not necessary

File: packages/main/src/components/channel/createchannel/createchannel.modeule.css [hunk 6]
```diff
@@ -0,0 +1,235 @@
+.CreateChannel__wrapper{
```
Comment (ickynavigator) [inline packages/main/src/components/channel/createchannel/createchannel.modeule.css:1]: you spelt this file wrong it should be `module`

File: packages/main/src/components/protected/sidebar/components/Room.jsx [hunk 7]
```diff
@@ -1,7 +1,7 @@
 import styles from "../styles/Drop.module.css";
 import RoomItem from "./RoomItem";
 
-export default function Room({ items, isOpen }) {
+export default function Room({ items, isOpen, setAddChannel }) {
```
Comment (ickynavigator) [inline packages/main/src/components/protected/sidebar/components/Room.jsx:4]: you added this prop but you are not using it

File: packages/main/src/components/protected/sidebar/components/Room.jsx [hunk 8]
```diff
@@ -21,6 +21,8 @@ export default function Room({ items, isOpen }) {
             );
           }
         })}
+
+      <li> </li>
```
Comment (ickynavigator) [inline packages/main/src/components/protected/sidebar/components/Room.jsx:25]: why did you add an empty `li`?

**Review by ickynavigator — CHANGES_REQUESTED** [review]

**Pull request closed without merging**